x-pack/platform/plugins/shared/observability_ai_assistant/server/routes/functions/route.ts

```diff
@@ -78,6 +81,44 @@ const getFunctionsRoute = createObservabilityAIAssistantServerRoute({
   },
 });
 
+const functionDatasetInfoRoute = createObservabilityAIAssistantServerRoute({
+  endpoint: 'GET /internal/observability_ai_assistant/functions/get_dataset_info',
+  params: t.type({
+    query: t.type({ index: t.string, connectorId: t.string }),
+  }),
+  security: {
+    authz: {
+      requiredPrivileges: ['ai_assistant'],
+    },
+  },
+  handler: async (resources) => {
+    const client = await resources.service.getClient({ request: resources.request });
+
+    const {
+      query: { index, connectorId },
+    } = resources.params;
+
+    const controller = new AbortController();
```

Comment:
is there a better way to do this? I understand this endpoint is only for testing but I am guessing we have an utility function somewhere to get an abort signal from the request.

Reply:
Isn't this just a matter of adding this?

```suggestion
    const controller = new AbortController();
    resources.request.events.aborted$.subscribe(() => {
      controller.abort();
    });    
```